C/common/datapoint.cpp

```diff
@@ -89,29 +89,42 @@ std::string DatapointValue::toString() const
 }
 
 /**
- * Delete the DatapointValue alongwith possibly nested Datapoint objects
+ * Delete the DatapointValue along with possibly nested Datapoint objects
  */
 void DatapointValue::deleteNestedDPV()
 {
 	if (m_type == T_STRING)
 	{
-		delete m_value.str;
-		m_value.str = NULL;
+		if (m_value.str) {
```

Comment:
delete is safe on NULL pointers. "The C++ language guarantees that delete p will do nothing if p is equal to NULL."